docs/libraries/typespec-client-generator-core/reference/linter.md

```diff
@@ -0,0 +1,31 @@
+---
+title: "Linter usage"
+toc_min_heading_level: 2
+toc_max_heading_level: 3
+---
+
+# Linter
+
+## Usage
+
+Add the following in `tspconfig.yaml`:
+
+```yaml
+linter:
+  extends:
+    - "@azure-tools/typespec-client-generator-core/all"
+```
+
+## RuleSets
+
+Available ruleSets:
+
+- `@azure-tools/typespec-client-generator-core/all`
+- `@azure-tools/typespec-client-generator-core/best-practices:csharp`
```

Comment:
What does best practice mean here?  Does it mean this is optional?  If so is there another category which is not optional?

Reply:
best practices is what I've called the ruleset for now, open to suggestions! I think the advice from the typespec team has been to use the word "best-practices", @timotheeguerin is this correct?

Reply:
Aren't these rulesets optional and can be opted out?

Reply:
oh I don't think we have a strong opinion on that name, we just have a placeholder `@typespec/best-practices` package that would contain misc linter to write good typespec.

I would say name this however makes the most sense. But I would note that I don't think this really should spend too much time on this as for all of azure they MUST use either the `data-plane` or `resource-manager` ruleset in teh spec repo. And I assume all those rules for all those language will be enabled.

I expect those more targeted ruleset might be more useful when we have the non azure version of tcgc(e.g. `@typespec/client`) where external customer might decide which language they care about.